include/openssl/macros.h

```diff
@@ -24,28 +24,57 @@
 # define NON_EMPTY_TRANSLATION_UNIT static void *dummy = &dummy;
 
 /*
- * Generic deprecation macro
-  *
-  * If OPENSSL_SUPPRESS_DEPRECATED is defined, then DECLARE_DEPRECATED
-  * becomes a no-op
+ * Generic deprecation macro, can be used as an attribute anywhere
```

Comment:
_anywhere_ is a bit strong ...

Reply:
So, "Generic deprecation macros", I guess...